api/src/main/java/jakarta/data/page/PageRequest.java

```diff
@@ -344,6 +344,32 @@ static <T> PageRequest<T> ofSize(int maxPageSize) {
      */
     PageRequest<T> previous();
 
+    /**
+     * <p>Returns a {@code PageRequest} requesting the next page
+     * if using cursor-based pagination. The given key must be the
+     * key of the last result on this page.</p>
+     *
+     * <p>Note that traversal of pages is usually done via
+     * {@link CursoredPage#nextPageRequest()}</p>
```

Comment:
I don't understand the benefit of adding the nextWithKey/previousWithKey methods.  Their own JavaDoc seems to imply that it should typically by done another way, and I agree with that.  Why are these needed?  I'd rather not add methods for a non-preferred way of doing things because it obscures the more useful methods.

Reply:
Because the only other way to do it is this disgusting monstrosity:

```java
			var _page =
					PageRequest.of(NaturalNumber.class)
							.sortBy((Iterable) pagination.sorts())
							.size(pagination.size())
							.page(pagination.page() + 1);
			return new CursoredPageRecord(_results.getResultList(), _cursors, _totalResults, pagination,
					_results.isLastPage() ? null : _page.afterKey(_results.getNextPage().getKey().toArray()),
					_results.isFirstPage() ? null : _page.beforeKey(_results.getPreviousPage().getKey().toArray()));
```

That's from my generated code, of course, but if I needed to do this, I guarantee you that at least some user will need to do it.

Reply:
I mean, the other option is to kill the `UnsupportedOperationException` that `Page.next()` throws, which is anyway of dubious value.

Reply:
If I'm reading your generated code correctly (I'm not sure what the types of some of the variables are and some of those methods don't exist in Jakarta Data), I think you could do this:
```java
long page = pagination.page();
return new CursoredPageRecord<NaturalNumber>(_results.getResultList(), _cursors, _totalResults, pagination,
                _results.isLastPage() ? null : pagination.page(page + 1).afterCursor(_cursors.get(_cursors.size() - 1)),
                _results.isFirstPage() ? null : pagination.page(page > 1 ? (page - 1) : 1).beforeCursor(_cursors.get(0)));
```

Reply:
Ah, thanks, let me try that.

Reply:
[So it's at least a _little_ weird that `nextPage()` throws when `page(page+1)` doesn't 😆]

Reply:
It's not _that_ bad. They actually just get a request for the next page by offset, with no cursor in it. So they don't get cursor-based pagination, but the _do_ get a perfectly legit next page, I think.

Right?

Reply:
>  I think you could do this

This works perfectly, thanks.